### DISCUSSION TITLE
Less import not working

### DISCUSSION THREAD

**Discussion by anarerdene** [vercel/next.js#23475] [Help]
### What version of Next.js are you using?

10.0.9

### What version of Node.js are you using?

v14.6.0

### What browser are you using?

brave

### What operating system are you using?

macOS

### How are you deploying your application?

next start

### Describe the Bug

Less global import not working

### Expected Behavior

updated next v9 to v10 and less not working

### To Reproduce

error - ./styles/main.less
Error: Cannot find module 'webpack/lib/node/NodeTemplatePlugin'
Require stack:
- /Works/app/program/node_modules/mini-css-extract-plugin/dist/loader.js
- /Works/app/program/node_modules/next/dist/compiled/webpack/bundle4.js
- /Works/app/program/node_modules/next/dist/compiled/webpack/webpack.js

in _app.js less imported globaly.

`import './styles/main.less'`

this is my next.config.js

`
```javascript
/* eslint-disable */
const withPlugins = require('next-compose-plugins');
const withCss = require('@zeit/next-css');
const withLess = require('@zeit/next-less');

// fix: prevents error when .less files are required by node
if (typeof require !== 'undefined') {
  require.extensions['.less'] = () => {};
  require.extensions['.css'] = (file) => {};
}


// Plugins
module.exports = withPlugins([
  [
    withCss,
    {
      lessLoaderOptions: {
        importLoaders: 1,
        sourceMap: true,
      },
    },
  ],
  [
    withLess,
    {
      lessLoaderOptions: {
        javascriptEnabled: true,
      },
    },
  ],

]);
````

**Comment by alexneo2003** [comment]
same issue

**Comment by hridoy43** [comment]
use this boilerplate guys. you have now less support with zero configuration. (It's an Antd boilerplate though and you have to reconfigure it if you don't want to use Antd in your project)

https://github.com/hridoy43/next-antd-boilerplate

**Comment by jamsinclair** [comment]
You'll need to install `webpack` as a dependency of your project (The solution is buried away in https://github.com/vercel/next.js/issues/21679).

I would recommend installing webpack v4 for stability with Next and the older plugins used by `@zeit/next-css`.
```shell
npm install -D webpack@^4.46.0
# Or 
yarn add --dev webpack@^4.46.0
```

**Comment by elado** [comment]
Try this: https://github.com/elado/next-with-less

**Reply by CracKerMe** [reply]
this help me a lot,tks

**Comment by nasatome** [comment]
I understand that this is a `npm install -D webpack@^4.46.0` solution but, is there any way to eliminate the use of the `@zeit/next-css` package, there are people that use simple CSS, and we simply use the LESS plugin because the framework we use requests it, in my particular case I use RsuiteJS https://github.com/rsuite/rsuite before I solved it by placing 

```
// next.config.js
const withLess = require('@zeit/next-less');
const withCSS = require('@zeit/next-css');
module.exports = withCSS(withLess({
  lessLoaderOptions: {
    javascriptEnabled: true
  }
})
);
```

and so I could import CSS and LESS, but now I see that the package is deprecated.
https://www.npmjs.com/package/@zeit/next-css

![image](https://user-images.githubusercontent.com/18271791/115611830-3447da00-a2b0-11eb-905f-0cd58ff0bf49.png)

**is there a way to import CSS and LESS or do I just change the way I import in my framework (Antd / Rsuite) and import the compiled CSS directly?**

**Reply by jamsinclair** [reply]
Totally agree @nasatome. As you've noticed the Vercel team (formerly Zeit) has deprecated `@zeit/next-css` and `@zeit/next-less` packages in favour of Next.js native CSS and SASS support.

Unless there's a strong reason they're unlikely to work on a solution for LESS 😿 

The good news is the community could find a solution and it seems like a few have cropped up. You can see @elado's one above, https://github.com/elado/next-with-less.
